shell/src/shell.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "shell/src/streams.h"
+#include "shell/src/parser.h"
+#include "shell/src/commands/command.h"
+
+#include <string>
+
+namespace shell {
+
+class Shell {
```

Comment:
I suggest to inherit `Shell` from `Command`, and to consider it like command (aka executable).
In general it has the same logic (`execute`), and as in bash it absolutely normal to call another bash instance, or run subshell/script.

Moreover - it costs us literally nothing but to preserve the same signature of `execute`. Benefits - subshells, scripts etc can be implemented only by addition `shell` into CommandManager